astropy/wcs/utils.py

```diff
@@ -71,6 +74,38 @@ def solar_system_body_representation_type(
     )
 
 
+def north_pole_angle(wcs, pixel=(0, 0), ddec=0.1 * u.arcsec):
```

Comment:
To avoid adding too much public API unless we think it is really needed, I think for now let's maybe not add this function to the API and instead just include the code below directly in ``add_compass``. In future if there is a strong need for a public function to find the angle specifically then we could consider splitting it out.

It would be good to also be able to specify what coordinate system one is finding the north of. If I am working with Galactic images I might want to show Galactic north. So having a way of passing in a frame to find the north of would be good.